northstar/src/runtime/repository.rs

```diff
@@ -14,44 +14,70 @@
 
 use std::{
     collections::HashMap,
+    ffi::OsStr,
     path::{Path, PathBuf},
-    time::SystemTime,
 };
 
-use super::{error::Error, Container, Name, RepositoryId, Version};
-use ed25519_dalek::PublicKey;
+use super::{
+    error::Error,
+    key::{self, PublicKey},
+    Container, RepositoryId,
+};
 use log::debug;
 use npk::npk::Npk;
 use tokio::fs;
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub(super) struct Repository {
     pub(super) id: RepositoryId,
     pub(super) dir: PathBuf,
     pub(super) key: Option<PublicKey>,
-    last_modified: Option<SystemTime>,
-    containers: HashMap<(Name, Version), PathBuf>,
+    pub(super) containers: HashMap<Container, PathBuf>,
+    pub(super) npks: HashMap<Container, Npk>,
```

Comment:
keeping `containers` and `npks` around seems redundant and probably should be consolidated